src/annotator/config/index.js

```diff
@@ -56,62 +66,98 @@ function configurationKeys(appContext) {
 }
 
 /**
- * Reads the Hypothesis configuration from the environment.
- *
- * @param {string[]} settingsKeys - List of settings that should be returned.
- * @param {Window} window_ - The Window object to read config from.
+ * The definition for configuration sources, their default values
+ * @return {ConfigDefinitionMap}
  */
-function configFrom(settingsKeys, window_) {
-  const settings = settingsFrom(window_);
-  const allConfigSettings = {
-    annotations: settings.annotations,
+function configDefinitions(settings) {
+  return {
+    annotations: {
+      valueFn: () => settings.annotations,
+    },
+    appType: {
+      valueFn: () =>
+        settings.hostPageSetting('appType', {
+          allowInBrowserExt: true,
+        }),
+    },
     // URL where client assets are served from. Used when injecting the client
     // into child iframes.
-    assetRoot: settings.hostPageSetting('assetRoot', {
-      allowInBrowserExt: true,
-    }),
-    branding: settings.hostPageSetting('branding'),
+    assetRoot: {
```

Comment:
This is untested because `assetRoot` does not belong to any app context because its part of the 'boot' context which is still TBD. Issue filed here. I could use some suggestions about what to do now.

https://github.com/hypothesis/client/issues/3474

Reply:
`assetRoot` is only used by the boot script, so you can ignore it here.